xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecDRMPRIME.cpp

```diff
@@ -143,13 +140,18 @@ bool CDVDVideoCodecDRMPRIME::Open(CDVDStreamInfo& hints, CDVDCodecOptions& optio
   if (pConfig && (pConfig->methods & AV_CODEC_HW_CONFIG_METHOD_HW_DEVICE_CTX) &&
       pConfig->device_type == AV_HWDEVICE_TYPE_DRM)
   {
-    CWinSystemGbm* winSystem = dynamic_cast<CWinSystemGbm*>(CServiceBroker::GetWinSystem());
-    if (av_hwdevice_ctx_create(&m_pCodecContext->hw_device_ctx, AV_HWDEVICE_TYPE_DRM,
-                               drmGetDeviceNameFromFd2(winSystem->GetDrm()->GetFileDescriptor()),
-                               nullptr, 0) < 0)
+    // todo: fix with proper device when dma-hints wayland protocol works
+    std::string device = "/dev/dri/renderD128";
```

Comment:
Please leave `drmGetDeviceNameFromFd2(winSystem->GetDrm()->GetFileDescriptor())` as the default, on my platform (with libmali) I currently only have `/dev/dri/card0` and no render device.

Reply:
ffmpeg will not use this device but it still needs it to initialize a hwdevice context.

Reply:
Or you could use a combo of `drmGetNodeTypeFromFd`, `drmGetRenderDeviceNameFromFd` and `drmGetDeviceNameFromFd2` to get the render node and fall back on `drmGetDeviceNameFromFd2` is there is no render node.

Reply:
@Kwiboo what is wrong with using the `KODI_RENDER_NODE` env variable?

Reply:
Nothing wrong with having the `KODI_RENDER_NODE` env as an override, but your are changing the default device to  `/dev/dri/renderD128` that do not exist on all platforms where hw decoding is working today.

Reply:
I just would like to avoid ifdeffing it if possible

Reply:
@Kwiboo could you propose a solution here please?

Reply:
@Kwiboo ping

Reply:
@Kwiboo please look over the latest change if possible, I would like to merge soon